### PR TITLE
docs: RDE Access Control Docs

### PR DESCRIPTION
**Pull request by kirilkadoncheva**
This commit contains the RDE Access Control documentation.

### CODE REVIEW

File: documentation/extensibility-platform/defined-entities/rde-access-control.md [hunk 1]
```diff
@@ -0,0 +1,751 @@
...
+            },
+            "privateField" : {
+              "type" : "string",
+              "x-vcloud-restricted" : "private"
+            }
+        }
+    }
+...
+```
+Secure fields are stored in the DB in an encrypted format (as a `base64-encoded` cipher string) and cannot be obtained via the Defined Entities API - they are either omitted or masked depending on the API version used.
```
Comment (iiliev2) [inline documentation/extensibility-platform/defined-entities/rde-access-control.md:409]: > cannot be obtained via the Defined Entities API in their decrypted form, apart from the dedicated endpoint `GET /cloudapi/1.0.0/entities/<entity-id>/fullContents`. This operation requires highest level of access to the RDE(even for users with `Admin Full Control` rights) and all its invocations are audited.

File: documentation/extensibility-platform/defined-entities/rde-access-control.md [hunk 2]
```diff
@@ -0,0 +1,751 @@
...
+
+When using API version `>= 38.0` secure fields are masked when returned in the API.Doing a `GET` on a defined entity with API version `38.1` for example, will return its `secure` fields masked (`******`) even if the user has Full Access Control to the defined entity instance.
+
+When using API version `< 38.0` secure fields are omitted when returned in the API.
+Doing a `GET` on a defined entity with API version `37.2` for example, will not return the entity's `secure` fields in the response even if the user has Full Access Control to the defined entity instance.
+
+### Accessing secure fields
+Behavior executions can access a defined entity instance's `secure` fields in a decrypted format if the user, who invoked the behavior has the required level of access to the defined entity (i.e. if the field is marked as `private` only users with Full Control access to the defined entity can access this field).
+
+A user with explicit Full Control Access to the defined entity (Admin Full Control users must also have a Full Control ACL to the specific defined entity) can access the secure fields of this entity via a dedicated API call:
```
Comment (iiliev2) [inline documentation/extensibility-platform/defined-entities/rde-access-control.md:419]: Remove these 3 lines. They should be moved closer to where we first introduce the `secure` concept and emphasised that this operation is "maximally" secured.

**Review by iiliev2 — APPROVED** [review]

File: documentation/extensibility-platform/defined-entities/rde-access-control.md [hunk 3]
```diff
@@ -0,0 +1,751 @@
...
+
+### Access Controls
+Access Control Lists (ACLs) are part of the access control mechanisms for Defined Entity Types and Defined Entity Instances. There are separate CRUD APIs for managing [Defined Entity Type ACLs](https://developer.vmware.com/apis/vmware-cloud-director/latest/type-access-controls/) and [Defined Entity ACLs](https://developer.vmware.com/apis/vmware-cloud-director/latest/access-controls/).
+
+An ACL entry contains the following fields:
+- `objectId` - The ID of the object this ACL refers to.
+- `accessLevelId` - The [access level](#access-levels) that this ACL entry grants.
+- `grantType` - The [type](#types-of-acls) of the ACL entry. Possible values are `MembershipAccessControlGrant` and `RightAccessControlGrant`.
+- `tenant` - (this field is filled automatically) The ID of the object's tenant. The ACL cannot be shared outside that tenant.
```
[the remark this answers is not in the document]
Comment (ttonchevvmware) [reply documentation/extensibility-platform/defined-entities/rde-access-control.md:37]: Perhaps:
The object and the principal must be within the ACL's tenant.

File: documentation/extensibility-platform/defined-entities/rde-access-control.md [hunk 4]
```diff
@@ -0,0 +1,751 @@
...
+
+An ACL entry contains the following fields:
+- `objectId` - The ID of the object this ACL refers to.
+- `accessLevelId` - The [access level](#access-levels) that this ACL entry grants.
+- `grantType` - The [type](#types-of-acls) of the ACL entry. Possible values are `MembershipAccessControlGrant` and `RightAccessControlGrant`.
+- `tenant` - (this field is filled automatically) The ID of the object's tenant. The ACL cannot be shared outside that tenant. 
+- `memberId` - (for Membership ACLs) the ID of the entity (user, organization, role) that this ACL is granted to.
+- `rightId` - (for Right ACLs) the ID of the right this ACL refers to.
+
+#### Types of ACLs
```
[the remark this answers is not in the document]
Comment (ttonchevvmware) [reply documentation/extensibility-platform/defined-entities/rde-access-control.md:41]: Valid comments. I would suggest leaving the RightAccessControlGrant discussion out of scope for now, though. It is mostly an internal matter at this stage.

**Review by ttonchevvmware — APPROVED** [review]
Approving as the documentation is in a good enough state.
We must address the comments additionally though.

**Review by theotonchev — APPROVED** [review]

**Pull request merged**